src/import/pull-common.c

```diff
@@ -637,6 +652,87 @@ int pull_job_restart_with_sha256sum(PullJob *j, char **ret) {
         return 1;
 }
 
+int signature_style_from_url(const char *url, SignatureStyle *ret, char **fn) {
+        _cleanup_free_ char *last = NULL;
+        SignatureStyle style;
+        int r;
+
+        assert(url);
+        assert(ret);
+        assert(fn);
+
+        /* Determines which kind of signature style is appropriate for this url */
+
+        r = import_url_last_component(url, &last);
+        if (r < 0)
+                return r;
+
+        style = signature_style_from_filename(last);
+        if (style >= 0) {
+                *fn = TAKE_PTR(last);
+                *ret = style;
+                return 0;
+        }
+
+        return style;
+}
+
+int pull_job_restart_with_signature(PullJob *j, char **ret) {
+        _cleanup_free_ char *last = NULL;
+        SignatureStyle style;
+        int r;
+
+        assert(j);
+
+        /* Generic implementation of a PullJobNotFound handler, that restarts the job requesting a different
+         * signature file. After the initial file, additional *.sha256.gpg, SHA256SUMS.gpg and SHA256SUMS.asc
+         * are tried in this order. */
+
+        r = signature_style_from_url(j->url, &style, &last);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine signature style of URL '%s': %m", j->url);
+
+        switch (style) {
+
+        case SIGNATURE_ASC_PER_DIRECTORY: /* Nothing to do anymore */
+                return 0;
```

Comment:
Please initialize `ret` with `NULL` on success.

Reply:
And as far as I understand the code, returning 0 is not success, that's 1.

Reply:
`ret` is nowhere else in the code initialized with `NULL` if you return 0, which is not success.

Reply:
Hence, `pull_job_restart_with_sha256sum()` needs to be fixed. If you'd like, please also fix it.